File: 2999-check-if-strings-can-be-made-equal-with-operations-i/check-if-strings-can-be-made-equal-with-operations-i.py
class Solution:
    def canBeEqual(self, s1: str, s2: str) -> bool:
        count = [0] * 26

        for i in range(len(s1)):
            count[ord(s1[i]) - ord('a')] += 1
            count[ord(s2[i]) - ord('a')] -= 1

        for c in count:
            if c != 0:
                return False

        if s1 == s2:
            return True

        if s1[0] == s2[2] and s2[0] == s1[2]:
            return True
        elif s1[1] == s2[3] and s2[1] == s1[3]:
            return True
        else:
            return False
        

File: 2999-check-if-strings-can-be-made-equal-with-operations-i/check-if-strings-can-be-made-equal-with-operations-i.java
class Solution {
    public boolean canBeEqual(String s1, String s2) {
        int[] count = new int[26];
        for (int i = 0; i < s1.length(); i++) {
            count[s1.charAt(i) - 'a']++;
            count[s2.charAt(i) - 'a']--;
        }
        for (int c : count) {
            if (c != 0) return false;
        }
        if (s1.equals(s2)) {
            return true;
        }
        if (s1.charAt(0) == s2.charAt(2) && s2.charAt(0) == s1.charAt(2) ) {
            return true;
        } else if (s1.charAt(1) == s2.charAt(3) && s2.charAt(1) == s1.charAt(3)) {
            return true;
        } else {
            return false;
        }
    }
}